### PR TITLE
Initial code to create a list of versions based on wp.org API.

### LINKED ISSUE #9: Version switching between tagged releases on .org
Introduce a screen which lists tagged versions/releases from the .org repository. 

Releases will have a radio button beside. Checking a radio and clicking 'switch' button will cause that version to be downloaded and installed. Once successful, a confirmation will be shown.

Look to WP Rollback for inspiration - this has something similar.

### PR DESCRIPTION
**Pull request by peterfabian**
Once done, it should fix #9.

Adding couple of functions to check versions fetched from wp.org API, once https://github.com/woocommerce/woocommerce-beta-tester/pull/19 is merged.

Based on those, an HTML code is generated to be able to pick version of WooCommerce (and later to switch to the picked version). Inspired by WP Rollback, as it works nicely.
TODOs: 
- [x] wait for bor0's changes to be merged
- [ ] better styling... add some CSS
- [x] more importantly, code should be split to admin module. 
- [ ] prevent conflict with version info modal

### CODE REVIEW

File: includes/class-wc-beta-tester.php [hunk 1]
```diff
@@ -294,4 +294,192 @@ public function upgrader_source_selection( $source, $remote_source, $upgrader )
...
+			$versions_html .= '</label>';
+			$versions_html .= '</li>';
+		}
+
+		$versions_html .= '</ul>';
+
+		return $versions_html;
+	}
+
+	// TODO: this needs css (and maybe a bit of JS), but the idea is to display Switch version button that displays modal which contains the actual form submit,
```
Comment (bor0) [inline includes/class-wc-beta-tester.php]: Something like this is done in #32

**Comment by claudiosanches** [comment]
Need fix some conflicts.

**Comment by peterfabian** [comment]
Resolved.

File: includes/class-wc-beta-tester.php [hunk 2]
```diff
@@ -391,4 +391,89 @@ public function get_version_information( $version ) {
...
 		return $github_data;
 	}
+
+	/**
+	 * Return true if version string is a beta version.
+	 *
+	 * @param string $version_str Version string.
+	 * @return bool
+	 */
+	protected static function is_beta_version( $version_str ) {
```
Comment (bor0) [inline includes/class-wc-beta-tester.php:449]: Do we need these to be static given the class is singleton?

File: includes/class-wc-beta-tester-version-picker.php [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+		// Loop through versions and output in a radio list.
+		foreach ( $tags as $tag_version ) {
+
+			$versions_html .= '<li class="wcbt-version-li">';
+			$versions_html .= '<label><input type="radio" value="' . esc_attr( $tag_version ) . '" name="wcbt_switch_to_version">' . $tag_version;
+
+			// Is this the current version?
+			if ( $tag_version === $this->current_version ) {
+				$versions_html .= '<span class="wcbt-current-version">' . esc_html__( 'Installed Version', 'woocommerce-beta-tester' ) . '</span>';
```
Comment (bor0) [inline includes/class-wc-beta-tester-version-picker.php]: Needs a `&nbsp;` or something :)

File: includes/class-wc-beta-tester-version-picker.php [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+												<td><span class="wcbt-new-version">{{ data.new_version }}</span></td>
+											</tr>
+											</tbody>
+										</table>
+
+										<div class="wcbt-notice">
+											<p><?php esc_html_e( 'Notice: We strongly recommend you perform the test on a staging site and create a complete backup of your WordPress files and database prior to performing a version switch. We are not responsible for any misuse, deletions, white screens, fatal errors, or any other issue arising from using this plugin.', 'woocommerce-beta-tester' ); ?></p>
+										</div>
+
+										<input type="submit" value="<?php esc_attr_e( 'Switch version', 'woocommerce-beta-tester' ); ?>" class="button-primary wcbt-go" id="wcbt-submit-version-switch" />
```
Comment (bor0) [inline includes/class-wc-beta-tester-version-picker.php]: I think it'd look a bit better if this and the next line went into `<footer />`?

File: assets/js/version-picker.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
+jQuery(function( $ ) {
+
+	//var instance = undefined;
```
Comment (bor0) [inline assets/js/version-picker.js]: Do we need this?

File: includes/class-wc-beta-tester.php [hunk 6]
```diff
@@ -391,4 +391,89 @@ public function get_version_information( $version ) {
...
+		$releases       = (array) $data->versions;
+		unset( $releases['trunk'] );
+
+		if ( 'beta' === $channel ) {
+			$releases = array_filter( $releases, array( __CLASS__, 'is_in_beta_channel' ), ARRAY_FILTER_USE_KEY );
+		} elseif ( 'rc' === $channel ) {
+			$releases = array_filter( $releases, array( __CLASS__, 'is_in_rc_channel' ), ARRAY_FILTER_USE_KEY );
+		} elseif ( 'stable' === $channel ) {
+			$releases = array_filter( $releases, array( __CLASS__, 'is_in_stable_channel' ), ARRAY_FILTER_USE_KEY );
+		};
```
Comment (bor0) [inline includes/class-wc-beta-tester.php]: Redundant semicolon

**Review by bor0 — APPROVED** [review]
Works for me in general. Just left a few minor comments.

File: includes/class-wc-beta-tester.php (hunk 2 above)
Comment (peterfabian) [reply includes/class-wc-beta-tester.php:449]: We don't need it, I just have a habit to mark functions that don't depend on/alter object's state as static. Is there a drawback to have this as static in php?

Comment (bor0) [reply includes/class-wc-beta-tester.php:449]: No drawbacks but I believe it's generally considered a code smell (mixture of static/non-static methods). I could be wrong though :)

File: includes/class-wc-beta-tester-version-picker.php (hunk 4 above)
Comment (peterfabian) [reply includes/class-wc-beta-tester-version-picker.php]: Yup, but if I add any longer text into footer, it covers the `<article>`, so maybe better to update the styling there?
I moved the buttons to the `<footer>`, though.

**Pull request merged**